### ISSUE TITLE
Cannot change cipher_suites

### ISSUE THREAD

**Issue by duj4** [grafana/agent#4698]
### What's wrong?

Our security team detected an vulnerability related to 3DES/DES(SSL Medium Strength Cipher Suites Supported, SWEET32) from the port that Grafana Agent is listening to, I tried to specify the secure cipher_suites (https://cs.opensource.google/go/go/+/refs/tags/go1.20.7:src/crypto/tls/cipher_suites.go;l=53) in config file but the vulnerability still exists.

### Steps to reproduce

1. Add cipher_suites in config file
2. Restart grafana-agent.service
3. Scan the os with Tenable.sc

### System information

Windows Server 2019

### Software version

0.35.2

### Configuration

```text
server:
  log_level: debug
  http_tls_config: &tls_config
    cipher_suites:
      - TLS_RSA_WITH_AES_128_CBC_SHA
      - TLS_RSA_WITH_AES_256_CBC_SHA
      - TLS_RSA_WITH_AES_128_GCM_SHA256
      - TLS_RSA_WITH_AES_256_GCM_SHA384
      - TLS_AES_128_GCM_SHA256
      - TLS_AES_256_GCM_SHA384
      - TLS_CHACHA20_POLY1305_SHA256
      - TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA
      - TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA
      - TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
      - TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
      - TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256
      - TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384
      - TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256
      - TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384
      - TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256
      - TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256
    windows_certificate_filter:
      server:
        system_store: LocalMachine
        store: My
        issuer_common_names:
          - "<common name>"
      client:
        issuer_common_names:
          - "<common name>"
  grpc_tls_config: *tls_config
```

### Logs

```text
Grafana-agent is running fine.
```

**Comment by duj4** [comment]
We mitigate the same issue in Linux by setting minimum_TLS to TLS1.3, but Windows Server 2019 does not support TLS1.3, this is the only way we could think about.

**Comment by duj4** [comment]
Does the comment mean changing cipher_suites is not supported:
https://github.com/grafana/agent/blob/main/pkg/server/tls.go#L216C27-L216C27

**Comment by tpaschalis** [comment]
The comment means that the CipherSuites cannot be updated dynamically (eg. changing them in the config and hitting the `/-/reload` endpoint).

I'm not sure what the Tenable.sc tool is looking for, but to make sure, have you enabled the `-server.http.enable-tls` command-line flag? Otherwise, the `http_tls_config` block is ignored.

**Comment by duj4** [comment]
> The comment means that the CipherSuites cannot be updated dynamically (eg. changing them in the config and hitting the `/-/reload` endpoint).
> 
> I'm not sure what the Tenable.sc tool is looking for, but to make sure, have you enabled the `-server.http.enable-tls` command-line flag? Otherwise, the `http_tls_config` block is ignored.

Thanks @tpaschalis , it is enabled and here is the details:
`-config.file="<config file location>" -server.http.enable-tls -server.http.address "<address:port>" -server.grpc.enable-tls -server.grpc.address "<grpc address:port>"`

**Comment by duj4** [comment]
@tpaschalis The Tenable.sc tool is used to scan the vulnerability across all of the opening ports on the target server and one of the areas it will cover is vulnerable cipher_suite.

**Comment by duj4** [comment]
Windows team disable 3DES and DES in OS level, but the vulnerability kept posting out on the port that GA is running.

Any further insight or update? @tpaschalis @marctc

**Comment by duj4** [comment]
We use nmap tool to scan the affected server and port 12345 again, same issue:
![微信截图_20230818140416](https://github.com/grafana/agent/assets/19278537/2b0e7136-d7a2-4064-ab8c-40b58cb1e0ce)

**Comment by duj4** [comment]
We also test this via openssl like below:
`openssl s_client -connect <target server:12345(whatever port GA is running on)> -cipher ECDHE-RSA-DES-CBC3-SHA(or whatever DES/3DES cipher_suite)`

If there is any response like `Secure Renegotiation IS NOT supported`, which means the cipher_suite is not functioning. Otherwise, it is affected.

**Comment by tpaschalis** [comment]
Thanks for the detailed information about how to test for this, taking a look.

**Comment by mattdurham** [comment]
With Windows dynamic ciphers are hard coded. This was done to have a wide availability of support. TLS is also locked down to 1.2 for the reason of OS support. IMO this would be an enhancement to add configuration to the windows cert store block.

**Comment by duj4** [comment]
@mattdurham which means that for now we cannot change cipher_suite in Windows?

**Comment by duj4** [comment]
@tpaschalis @mattdurham May I know if there is any update on this issue?

**Comment by mattdurham** [comment]
There is currently no update nor ETA on if/when it will be changed.

**Comment by duj4** [comment]
Thanks @mattdurham , is there any workaround we could bypass this issue?

**Comment by duj4** [comment]
Well, we tried windows_exporter along with web-config.yml and set minimum version of TLS to 1.3, the issue is gone.

**Comment by duj4** [comment]
Hi Team,
We found this issue only occur when integrating with Windows cert store.

**Comment by mattdurham** [comment]
We are going to be adding the ability to specify tls version in the upcoming version in PR https://github.com/grafana/agent/pull/5176

**Comment by duj4** [comment]
@mattdurham thanks for the update, looking forward to it.

**Comment by rfratto** [comment]
Closing as completed as #5176 has been merged. The functionality will be available in the upcoming 0.37 release.
